.github/workflows/appimage.yml

```diff
@@ -26,12 +26,25 @@ jobs:
     timeout-minutes: 30
 
     steps:
-    - name: Check out code in viam-cartographer directory
+    - name: Check out main code
+      if: github.event_name == 'workflow_dispatch' || github.event_name == 'push'
       uses: actions/checkout@v3
       with:
         submodules: recursive
         path: viam-cartographer
+      
+    - name: Check out PR branch code
+      if: github.event_name == 'pull_request_target'
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        submodules: recursive
+        path: viam-cartographer
 
+    - name: make clean
+      run: |
+        sudo -u testbot bash -lc 'cd viam-cartographer && make clean'
```

Comment:
Why do we need to add this step? Is this fixing a bug?

Reply:
Best practice, but _technically_ should always be a clean checkout under a github runner.